source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -225,22 +227,14 @@ RedisCluster::RedisDiscoverySession::RedisDiscoverySession(
 // Convert the cluster slot IP/Port response to and address, return null if the response
 // does not match the expected type.
 Network::Address::InstanceConstSharedPtr
-RedisCluster::RedisDiscoverySession::RedisDiscoverySession::ProcessCluster(
+RedisCluster::RedisDiscoverySession::RedisDiscoverySession::processClusterByIP(
     const NetworkFilters::Common::Redis::RespValue& value) {
-  if (value.type() != NetworkFilters::Common::Redis::RespType::Array) {
-    return nullptr;
-  }
-  auto& array = value.asArray();
-
-  if (array.size() < 2 || array[0].type() != NetworkFilters::Common::Redis::RespType::BulkString ||
-      array[1].type() != NetworkFilters::Common::Redis::RespType::Integer) {
-    return nullptr;
-  }
 
+  const auto& array = value.asArray();
```

Comment:
If this function now always expects the value to be an array, should the function signature be changed to accept the array itself? I think that would make the assumptions made by this function a bit clearer?

Or should we keep the extra validation in place? Could a misbehaving upstream cause an issue here if it responds with the wrong data format?

Reply:
We now have a `validateCluster` method which performs the validation, before calling this method. So, changing it to accept the array itself is a good point. I've changed it to reflect it.